zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -308,64 +335,172 @@ benchmarks! {
         let (caller, market_id) = create_market_common::<T>(
             MarketCreation::Permissionless,
             MarketType::Categorical(a.saturated_into()),
-            ScoringRule::CPMM
+            ScoringRule::CPMM,
+            None,
         )?;
         let amount = BASE * 1_000;
     }: _(RawOrigin::Signed(caller), market_id, amount.saturated_into())
 
     // Beware! We're only benchmarking categorical markets (scalar market creation is essentially
     // the same).
     create_market {
-        let (caller, oracle, period, metadata, creation) =
+        let m in 0..63;
+
+        let (caller, oracle, metadata, creation) =
             create_market_common_parameters::<T>(MarketCreation::Permissionless)?;
+
+        let range_end = T::MaxSubsidyPeriod::get();
+        let period = MarketPeriod::Timestamp(T::MinSubsidyPeriod::get()..range_end);
+
+        for i in 0..m {
+            MarketIdsPerCloseTimeFrame::<T>::try_mutate(
+                Pallet::<T>::calculate_time_frame_of_moment(range_end),
+                |ids| ids.try_push(i.into()),
+            ).unwrap();
+        }
     }: _(RawOrigin::Signed(caller), oracle, period, metadata, creation,
             MarketType::Categorical(T::MaxCategories::get()),
             MarketDisputeMechanism::SimpleDisputes, ScoringRule::CPMM)
 
-    deploy_swap_pool_for_market {
+    deploy_swap_pool_for_market_future_pool {
         let a in (T::MinCategories::get().into())..T::MaxCategories::get().into();
+        let o in 0..63;
+
+        let range_start: MomentOf<T> = 100_000u64.saturated_into();
+        let range_end: MomentOf<T> = 1_000_000u64.saturated_into();
         let (caller, market_id) = create_market_common::<T>(
             MarketCreation::Permissionless,
             MarketType::Categorical(a.saturated_into()),
-            ScoringRule::CPMM
+            ScoringRule::CPMM,
+            Some(MarketPeriod::Timestamp(range_start..range_end)),
         )?;
+
+        assert!(
```

Comment:
[nitpick - no need to change this now!] You can also use `asset_le!` for this. That has the advantage of formatting to something more readable.